src/cider/tests/benchmark/ArrowDataReader.h

```diff
@@ -0,0 +1,162 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <arrow/api.h>
+#include <arrow/csv/api.h>
+#include <arrow/io/api.h>
+#include <arrow/io/file.h>
+#include <parquet/arrow/reader.h>
+#include "exec/plan/parser/TypeUtils.h"
+#include "tests/utils/ArrowArrayBuilder.h"
+#include "tests/utils/Utils.h"
+
+namespace {
+
+#define constructArrowFixedColumn(s_type)                         \
+  {                                                               \
+    for (int j = 0; j < chunk_arrays->num_chunks(); j++) {        \
+      auto arrow_array = chunk_arrays->chunk(j);                  \
+      builder.addColumn(name,                                     \
+                        s_type,                                   \
+                        arrow_array->null_bitmap_data(),          \
+                        arrow_array->data()->buffers[1]->data()); \
+    }                                                             \
+    break;                                                        \
+  }
+
+#define constructArrowVariColumn(s_type)                          \
+  {                                                               \
+    for (int j = 0; j < chunk_arrays->num_chunks(); j++) {        \
+      auto arrow_array = chunk_arrays->chunk(j);                  \
+      builder.addColumn(name,                                     \
+                        s_type,                                   \
+                        arrow_array->null_bitmap_data(),          \
+                        arrow_array->data()->buffers[1]->data(),  \
+                        arrow_array->data()->buffers[2]->data()); \
+    }                                                             \
+    break;                                                        \
+  }
+
+void constructArrowArray(ArrowArrayBuilder& builder,
+                         std::shared_ptr<arrow::ChunkedArray> chunk_arrays,
+                         const std::string& name) {
+  auto tag = chunk_arrays->type()->id();
+  switch (tag) {
+    case arrow::Type::type::INT32:
+      constructArrowFixedColumn(CREATE_SUBSTRAIT_TYPE(I32));
+    case arrow::Type::type::INT64:
+      constructArrowFixedColumn(CREATE_SUBSTRAIT_TYPE(I64));
+    case arrow::Type::type::FLOAT:
+      constructArrowFixedColumn(CREATE_SUBSTRAIT_TYPE(Fp32));
+    case arrow::Type::type::DOUBLE:
+      constructArrowFixedColumn(CREATE_SUBSTRAIT_TYPE(Fp64));
+    case arrow::Type::type::STRING:
+      constructArrowVariColumn(CREATE_SUBSTRAIT_TYPE(Varchar));
+  }
+}
+}  // namespace
+
+class ArrowDataReader {
+ public:
+  ArrowDataReader(std::string file_name, std::vector<std::string>& col_name)
```

Comment:
`col_name` should be all columns? Can I only read the 1st, 3rd column via this API? Please give some comments.